source/ide/ide_methods.bas

```diff
@@ -20604,29 +20620,38 @@ SUB ExportCodeAs (docFormat$)
     WEND
     GOSUB CloseCodeBlock
     '----------
-    ideerror = 7
-    OPEN idepath$ + idepathsep$ + pNam$ + ext$ FOR OUTPUT AS #151
-    ideerror = 1
-    PRINT #151, LEFT$(eTxt$, ePos& - 1);
-    CLOSE #151
     PCOPY 2, 3
+    SELECT CASE LCASE$(docFormat$)
+        CASE "html", "rich"
+            ideerror = 7
+            OPEN idepath$ + idepathsep$ + pNam$ + ext$ FOR OUTPUT AS #151

```

Comment:
Is there a reason not to use `FREEFILE` here? If we did then we wouldn't have to worry about 151 already being in use.

Reply:
Yep, found that file 151 is used for quick "Write in place" operations in the qb64pe sources, hence immediate OPEN-WRITE-CLOSE constructs like saving changed source, I just adapted to it, as writing the export file is practically the same operation.